File: check-db.py
import mysql.connector
import os

# Load DB config from environment variables or .env file
DB_HOST = os.getenv('DB_HOST', '127.0.0.1')
DB_PORT = int(os.getenv('DB_PORT', 3306))
DB_DATABASE = os.getenv('DB_DATABASE', 'job_portal')
DB_USERNAME = os.getenv('DB_USERNAME', 'root')
DB_PASSWORD = os.getenv('DB_PASSWORD', '')

print('Connecting to database...')
conn = mysql.connector.connect(
    host=DB_HOST,
    port=DB_PORT,
    user=DB_USERNAME,
    password=DB_PASSWORD,
    database=DB_DATABASE
)
cursor = conn.cursor()

print('Checking database tables...')
cursor.execute('SHOW TABLES')
tables = cursor.fetchall()
for (table_name,) in tables:
    if 'job' in table_name:
        print(f'Found: {table_name}')

print('\nChecking jobs table structure...')
cursor.execute('DESC jobs')
columns = cursor.fetchall()
for col in columns:
    print(f'  - {col[0]} ({col[1]})')

cursor.close()
conn.close()

File: check-db.js

// Ported from check-db.php
// Checks for tables containing 'job' and prints structure of 'jobs' table.
import mysql from 'mysql2/promise';

async function main() {
	const connection = await mysql.createConnection({
		host: process.env.DB_HOST || '127.0.0.1',
		port: process.env.DB_PORT || 3306,
		user: process.env.DB_USERNAME || 'root',
		password: process.env.DB_PASSWORD || '',
		database: process.env.DB_DATABASE || 'job_portal',
	});

	console.log('Checking database tables...');
	const [tables] = await connection.execute('SHOW TABLES');
	tables.forEach(row => {
		const tableName = Object.values(row)[0];
		if (tableName.includes('job')) {
			console.log(`Found: ${tableName}`);
		}
	});

	console.log('\nChecking jobs table structure...');
	const [columns] = await connection.execute('DESC jobs');
	columns.forEach(col => {
		console.log(`  - ${col.Field} (${col.Type})`);
	});

	await connection.end();
}

main().catch(err => { console.error(err); process.exit(1); });
